### PR TITLE
Add Destination Address whitelisting

### PR DESCRIPTION
**Pull request by shamilkhan**
Added ability to select destination address from Polkadot-JS Web Extension or paste it manually.

### CODE REVIEW

File: src/Components/Pages/TransferPage.tsx [hunk 1]
```diff
@@ -424,6 +430,8 @@ const TransferPage = () => {
     tokenSymbol: "",
   });
 
+  const isDestinationBridgeAddressList = useMemo(() => Boolean(destinationBridge.addresses.length), [destinationBridge.addresses]);
```
Comment (MRamanenkau) [inline src/Components/Pages/TransferPage.tsx]: The name is confusing. `isDestinationAddressListExists` looks much better

File: src/Components/Pages/TransferPage.tsx [hunk 2]
```diff
@@ -411,6 +416,7 @@ const TransferPage = () => {
     checkSupplies,
   } = useChainbridge();
 
+  const [isPasteDestinationAddressManually, setIsPasteDestinationAddressManually] = useState(false);
```
Comment (MRamanenkau) [inline src/Components/Pages/TransferPage.tsx]: The name is confusing. What do you think about `shouldPasteDestionationAddressManually`?

File: src/Components/Pages/TransferPage.tsx [hunk 3]
```diff
@@ -750,20 +758,36 @@ const TransferPage = () => {
...
+                          />
+                      )}
+                      {isDestinationBridgeAddressList && (
+                          <CheckboxInput
+                              className={classes.destinationAddressCheckbox}
+                              label="Paste Address Manually"
+                              value={isPasteDestinationAddressManually}
+                              onChange={() => setIsPasteDestinationAddressManually(!isPasteDestinationAddressManually)}
+                          />
+                      )}
```
Comment (MRamanenkau) [inline src/Components/Pages/TransferPage.tsx]: The logic is a bit weird. Can we use a component that can be used as an "input" and "select" at the same time

Comment (shamilkhan) [reply src/Components/Pages/TransferPage.tsx]: I made this decision based on two primary reasons:
1. The UI kit does not currently offer a component that allows for both selection and input simultaneously.
2. While we use Whitelisting (via the Polkadot-JS Web Extension) for transfers from EVM to Substrate, transfers from Substrate to EVM still require manual input.

File: src/Components/Pages/TransferPage.tsx (hunk 2 above)
Comment (shamilkhan) [reply src/Components/Pages/TransferPage.tsx]: Yes. It's deferentially better.

File: src/Components/Pages/TransferPage.tsx (hunk 3 above)
Comment (MRamanenkau) [reply src/Components/Pages/TransferPage.tsx]: Point two can usually be handled by such a component but if the UI kit doesn't have it let's keep your solution

**Review by MRamanenkau — DISMISSED** [review]

**Review by MRamanenkau — APPROVED** [review]

**Pull request merged**